projects/habitat_ovmm/evaluator.py

```diff
@@ -168,12 +129,282 @@ def stop():
 
         envs.close()
-        self.write_results(episode_metrics)
+
+        aggregated_metrics = self._aggregate_metrics(episode_metrics)
+        self._write_results(episode_metrics, aggregated_metrics)
+
+        return aggregated_metrics
+
+    def _aggregate_metrics(self, episode_metrics: Dict[str, Any]) -> Dict[str, float]:
+        """Aggregates metrics tracked by environment."""
+        aggregated_metrics = defaultdict(list)
+        metrics = set(
+            [
+                k
+                for metrics_per_episode in episode_metrics.values()
+                for k in metrics_per_episode
+                if k != "goal_name"
+            ]
+        )
+        for v in episode_metrics.values():
+            for k in metrics:
+                if k in v:
+                    aggregated_metrics[f"{k}/total"].append(v[k])
+
+        aggregated_metrics = dict(
+            sorted(
+                {
+                    k2: v2
+                    for k1, v1 in aggregated_metrics.items()
+                    for k2, v2 in {
+                        f"{k1}/mean": np.mean(v1),
+                        f"{k1}/min": np.min(v1),
+                        f"{k1}/max": np.max(v1),
+                    }.items()
+                }.items()
+            )
+        )
+
+        return aggregated_metrics
+
+    def _write_results(
+        self, episode_metrics: Dict[str, Dict], aggregated_metrics: Dict[str, float]
+    ) -> None:
+        """Writes metrics tracked by environment to a file."""
+        with open(f"{self.results_dir}/aggregated_results.json", "w") as f:
+            json.dump(aggregated_metrics, f, indent=4)
+        with open(f"{self.results_dir}/episode_results.json", "w") as f:
+            json.dump(episode_metrics, f, indent=4)
+
+    def local_evaluate(
+        self, agent, num_episodes: Optional[int] = None
+    ) -> Dict[str, float]:
+        """
+        Evaluates the agent in the local environment.
+
+        :param agent: agent to be evaluated in environment.
+        :param num_episodes: count of number of episodes for which the evaluation should be run.
+        :return: dict containing metrics tracked by environment.
+        """
+
+        env_num_episodes = self._env.number_of_episodes
+        if num_episodes is None:
+            num_episodes = env_num_episodes
+        else:
+            assert num_episodes <= env_num_episodes, (
+                "num_episodes({}) is larger than number of episodes "
+                "in environment ({})".format(num_episodes, env_num_episodes)
+            )
+
+        assert num_episodes > 0, "num_episodes should be greater than 0"
+
+        episode_metrics: Dict = {}
+
+        count_episodes: int = 0
+
+        pbar = tqdm(total=num_episodes)
+        while count_episodes < num_episodes:
+            observations, done = self._env.reset(), False
+            current_episode = self._env.get_current_episode()
+            agent.reset_vectorized([current_episode])
```

Comment:
Hi @yvsriram, our evaluation code should work for all the agents that implement [home_robot.core.abstract_agent.Agent](https://github.com/facebookresearch/home-robot/blob/e6e81b4f162f163fec5ba614a96c599992eff326/src/home_robot/home_robot/core/abstract_agent.py#L13 ) class. 

Can I replace `agent.reset_vectorized([current_episode])` with just `agent.reset()`? Otherwise, we require all the participants agents to implement this function.

Reply:
`OpenVocabManipAgent` is a subclass of `ObjectNavAgent` and `ObjectNavAgent` has `reset` implemented as follows:
```python
    def reset(self):
        """Initialize agent state."""
        self.reset_vectorized()
        self.planner.reset()
```
Can we just reuse this inherited method? We can update the base abstract class but usually reset should not take any arguments.

Reply:
- Yes, I think we shouldn't pass the entire `episode` here, as the user may instead use fields from episode, which won't work in real world (cc: @JHurricane96)
- But without this: the planner snapshots for all scenes get saved in the same folder. This will make debugging from the codebase difficult. This is what happens in ObjectNavAgent as well. The planner snapshots keep getting overriden.
- I think we should update the planner's vis_dir inside`eval_baselines_agent.py` instead after we call `reset`.